cmd/syft/internal/options/source.go

```diff
@@ -20,7 +22,8 @@ type sourceConfig struct {
 }
 
 type fileSource struct {
-	Digests []string `json:"digests" yaml:"digests" mapstructure:"digests"`
+	Digests      []string `json:"digests" yaml:"digests" mapstructure:"digests"`
+	MaxLayerSize string   `json:"max-layer-size" yaml:"max-layer-size" mapstructure:"max-layer-size"`
```

Comment:
Hey, @tomersein I think we should add this configuration to the `imageSource` ([see here](https://github.com/anchore/syft/blob/main/cmd/syft/internal/options/source.go#L36C6-L38)), instead of the `fileSource`. Other than that, I think this PR is probably good. Thanks 👍

Reply:
hi @kzantow the image-souce doesn't work when i put it inside, only filesource.
does it make sense?

Reply:
You need to add a `PostLoad` hook for the `imageSource`; I think it would look like this:
```
func (c *imageSource) PostLoad() error {
	if c.MaxLayerSize != "" {
		perFileReadLimit, err := humanize.ParseBytes(c.MaxLayerSize)
		if err != nil {
			return err
		}
		stereoscopeFile.SetPerFileReadLimit(int64(perFileReadLimit))
	}
	return nil
}
```
You could ensure it's the right signature with:
```
var _ clio.PostLoader = (*imageSource)(nil)
```